src/phases/query/Multiplanner.yml

```diff
@@ -147,3 +147,95 @@ Hide14IndexesTemplate:
         - {OperationName: RunCommand, OperationCommand: {collMod: *coll2, index: {hidden: true, name: index14}}}
         - {OperationName: RunCommand, OperationCommand: {collMod: *coll2, index: {hidden: true, name: index15}}}
         - {OperationName: RunCommand, OperationCommand: {collMod: *coll2, index: {hidden: true, name: index16}}}
+
+fraction: &frac {^NumExpr: {withExpression: "resultCount / docCount", andValues: {
+                      resultCount: &resultCount {^Parameter: {Name: "resultCount", Default: {unused: "Invalid result count."}}},
+                      docCount: {^Parameter: {Name: "docCount", Default: {unused: "Invalid docCount."}}}}}}
+
+FindWithNExpectedResults:
+  Name: FindWithNExpectedResults
+  Type: RunCommand
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: {^Parameter: {Name: "active", Default: {unused: "Invalid phase number."}}}
+      NopInPhasesUpTo: {^Parameter: {Name: "nopInPhasesUpTo", Default: {unused: "Invalid phase number."}}}
+      PhaseConfig:
+        Repeat: {^Parameter: {Name: "repeat", Default: {unused: "Invalid repeat count."}}}
+        Database: {^Parameter: {Name: "database", Default: {unused: "Invalid database name."}}}
+        Operation: 
+          {OperationName: RunCommand, OperationCommand: {
+            find: {^Parameter: {Name: "collection", Default: {unused: "Invalid collection name."}}},
+            let: {
+              lowBound: {$multiply: [{$subtract: [1, *frac]}, {^RandomDouble: {distribution: uniform, min: 0.0, max: 1.0}}]}, 
+              size: *frac
+            },
+            filter: {$expr: {$and: [
+                      # Confirm that lowBound <= x1 < lowBound + size
```

Comment:
Did you check with explain() that optimizer is smart enough to constant fold this and extract correct index bounds?

Reply:
Yep, everything looks good in that regard.